File: ISchemm.OpenTokFs.RequestTypes/BroadcastStartRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace ISchemm.OpenTokFs.RequestTypes
{
    public interface IBroadcastStartRequest
    {
        string SessionId { get; }
        string LayoutType { get; }
        string LayoutStylesheet { get; }
        TimeSpan Duration { get; }
        bool Hls { get; }
        IEnumerable<IRtmpDestination> Rtmp { get; }
        string Resolution { get; }
    }

    public class BroadcastStartRequest : IBroadcastStartRequest
    {
        public string SessionId { get; }
        public string LayoutType { get; set; }
        public string LayoutStylesheet { get; set; }
        public TimeSpan Duration { get; set; }
        public bool Hls { get; set; }
        public IEnumerable<IRtmpDestination> Rtmp { get; set; }
        public string Resolution { get; set; }

        public BroadcastStartRequest(string sessionId)
        {
            this.SessionId = sessionId ?? throw new ArgumentNullException(nameof(sessionId));
            this.LayoutType = "bestFit";
            this.LayoutStylesheet = null;
            this.Duration = TimeSpan.FromHours(2);
            this.Hls = false;
            this.Rtmp = Enumerable.Empty<IRtmpDestination>();
            this.Resolution = "640x480";
        }
    }
}

File: RequestTypes/BroadcastStartRequest.fs
namespace OpenTokFs.RequestTypes

open System

/// <summary>
/// An object that provides parameters for starting an OpenTok broadcast, using reasonable defaults.
/// Note that neither HLS nor RTMP is enabled by default; you will have to turn one or both of them on.
/// </summary>
type BroadcastStartRequest(sessionId: string) =
    member __.SessionId = sessionId
    member val Layout: VideoLayout = VideoLayout.BestFit with get, set
    member val Duration: TimeSpan = TimeSpan.FromHours 2.0 with get, set
    member val Hls: bool = false with get, set
    member val Rtmp: RtmpDestination seq = Seq.empty with get, set
    member val Resolution: string = "640x480" with get, set